### PR TITLE
🔪 termops.maskAddress

### PR DESCRIPTION
**Pull request by apendleton**
### Context
This branch tweaks boundaries-plus to delete the maskAddress function and corresponding tests, and replace them with a property passthrough mechanism. In current master, each phrasematch for an address index potentially represents a hypothesis for which token in the query represents an address number, and we perform a geocode based on that hypothesis, but we don't actually store it as we proceed, so by the time we get to verifymatch, we don't remember what it is anymore, and have to guess, and sometimes we guess wrong. Some changes around token replacement in boundaries-plus and other repos seem to make the incorrect guessing somewhat more likely in some circumstances, especially in Japanese when using a newly-possible non-global approach to Japanese number standardization, which has forced our hand in dealing with this now.

The approach is basically to stop guessing, and instead remember what we did in the first place (as proposed in #648 and elsewhere). Steps:

* numTokenize now stores what the original number and position were of the number it's replacing
* permutations checks to see as it generates each permutation if that permutation contains stored address metadata supplied by numTokenize
* phrasematch copies the address info off of the permutation and onto the fuzzy match response, and resets the position to be the true position (using the `owner` map) if necessary
* it also now copies the address metadata on gap expansion, if necessary
* in spatialmatch, after stacking, for each stack before it's coalesced, any address metadata associated with any element in the stack is stored, keyed by index
* after coalesce, the indexes in the response elements in the stack are used to re-retrieve the address metadata and reattach it to the mask
* in verifymatch, rather than calling maskAddress, we just use the address data passed in on the spatialmatch object

Incidentally this also lays some groundwork that I think will make #827 easier.
 
### Summary of Changes
- [x] add logic to store and pass through number and position
- [x] delete maskAddress and corresponding unit tests
- [x] update other tests with new additional expected metadata
- [x] add new Japanese address test that fails on boundaries-plus because maskAddress guessed wrong

### Next Steps
nope

cc @mapbox/search

### CODE REVIEW

File: lib/geocoder/spatialmatch.js [hunk 1]
```diff
@@ -253,10 +260,10 @@ function allowed(stacks, options) {
     return filtered;
 }
 
-function stackByIdx(stack) {
-    const byIdx = {};
+function getStackByIdx(stack) {
```
Comment (apendleton) [inline lib/geocoder/spatialmatch.js]: stackByIdx referred to a function in some parts of the file and the results you got from calling the function in others, which was confusing. Made the data always be stackByIdx and the function by getStackByIdx. (Also made it return a map because yay maps, and to make it more consistent with newer address stuff.)

**Comment by apendleton** [comment]
This is failing codecov because of a single line that I made a change to that wasn't covered in the original branch either, and I have no idea what it does (but I had to fix it because I changed the type of an affected variable). Since this is not a PR into master, I'm inclined not to worry about it.

File: lib/geocoder/verifymatch.js [hunk 2]
```diff
@@ -3,7 +3,6 @@ const addressItp = require('./addressitp');
 const addressCluster = require('./addresscluster');
 const queue = require('d3-queue').queue;
 const context = require('./context');
-const termops = require('../text-processing/termops');
```
Comment (miccolis) [inline lib/geocoder/verifymatch.js]: This make me unreasonably happy.
